### ISSUE TITLE
Drag selection tool target is set to 0,0 sporadically when zoomed out

### ISSUE THREAD

**Issue by oli414** [OpenRCT2/OpenRCT2#11616]
**OS:** Windows 10
**Version:** 0.2.6
**Commit/Build:** b5fed0d

The drag selection tool (from the plugin system) has the second target position set to 0,0 when hovering over slopes in a fully zoomed out map.

**Steps to reproduce:**
1. Open a plugin that uses a draggable selection area
2. Zoom all the way out
3. Make a selection
4. Drag over a landscape with height differences
5. The selection will sometimes change from the starting position to 0,0 instead of the starting position to the mouse position

**Screenshots / Video:**
https://i.gyazo.com/753b56f0002254baab4a307edd496795.mp4

**Comment by Broxzier** [comment]
While developing the horizontal-clipping feature for the cut-away view tool I bumped into this very same issue. This was fixed by only updating the selection when the mouse is hovering over a surface element, otherwise keep its previous selection. The same behaviour can be triggered by hovering over the blank void outside the map.

https://github.com/OpenRCT2/OpenRCT2/blob/935cfe90fc111eeb170a9e65227fe5a25f4a4b00/src/openrct2-ui/windows/ViewClipping.cpp#L322-L330
